core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -220,12 +220,13 @@ public Expression residual() {
     }
 
     public boolean isAdjacent(SplitScanTask other) {
-      return (other != null) &&
-          (this.file().equals(other.file())) &&
-          (this.offset + this.len == other.offset);
+      return other != null &&
+          this.file().equals(other.file()) &&
+          this.offset + this.len == other.offset;
     }
   }
 
+  @SuppressWarnings("MixedMutabilityReturnType")
```

Comment:
This seems unrelated to the parentheses change. What is the purpose of this?

Reply:
I will fixing all static analysis warnings with this PR.